File: moling-server/app/schemas/health.py
"""墨灵 (Moling) — Health Alert Pydantic Schemas."""

from __future__ import annotations

from datetime import datetime
from typing import Optional, List

from pydantic import BaseModel, Field


class HealthAlertResp(BaseModel):
    """Health alert response."""

    id: int = Field(..., description="告警 ID")
    rule: str = Field(..., description="触发的规则名称")
    title: str = Field(..., description="告警标题")
    detail: str = Field(..., description="告警详情")
    severity: str = Field(..., description="严重程度")
    is_active: bool = Field(default=True, description="是否活跃")
    checked_at: Optional[datetime] = Field(default=None, description="最后检查时间")
    created_at: datetime = Field(..., description="创建时间")

    model_config = {"from_attributes": True}


class HealthAlertItem(BaseModel):
    """单个健康告警项（用于工作台健康告警横幅）。"""

    rule: str = Field(..., description="触发的规则名称（R1/R2/R3）")
    title: str = Field(..., description="告警标题")
    detail: str = Field(..., description="告警详细信息")

    model_config = {"from_attributes": True}


class HealthCheckResp(BaseModel):
    """项目健康检查响应（用于 GET /projects/:pid/health）。"""

    alerts: List[HealthAlertItem] = Field(..., description="健康告警列表")
    checked_at: datetime = Field(..., description="检查时间")

    model_config = {"from_attributes": True}

File: moling-web/src/mock/health.ts
/* ============================================
   墨灵 (Moling) — Mock Health Alerts
   ============================================ */

import type { HealthAlert } from "@/lib/types";

export const mockHealthAlerts: HealthAlert[] = [
  {
    id: "1",
    rule: "plot_hole",
    title: "剧情漏洞检测",
    detail: "第3章中，林星辰从练气一层直接击败了三阶妖兽，战力跨度不合理。建议增加修炼突破的情节铺垫。",
    severity: "warning",
    is_active: true,
  },
  {
    id: "2",
    rule: "character_consistency",
    title: "角色一致性提醒",
    detail: "苏月瑶在第2章中表现出对修真一无所知，但在第3章却能认出上古阵法秘纹。请检查角色认知的前后一致性。",
    severity: "info",
    is_active: true,
  },
  {
    id: "3",
    rule: "timeline_conflict",
    title: "时间线冲突",
    detail: `第3章的试炼描述中出现了"月光"，但在第1章的时间设定中当天应为新月。时间描写存在矛盾。`,
    severity: "critical",
    is_active: true,
  },
];
